src/elm/Session/LoggedIn.elm

```diff
@@ -177,8 +169,8 @@ type FeatureStatus
     | FeatureLoading
 
 
-initModel : Shared -> Auth.Model -> Eos.Name -> Symbol -> String -> Model
-initModel shared authModel accountName selectedCommunity authToken =
+initModel : Shared -> Maybe String -> Eos.Name -> Symbol -> String -> Model
+initModel shared maybePrivateKey_ accountName selectedCommunity authToken =
```

Comment:
we could introduce a type for private key instead of using a `String`

Reply:
I actually introduced a `type alias PrivateKey = String` later on and forgot to change stuff to use it (will do so). I don't think we need a `type PrivateKey` for this though, it would just mean more wrapping/unwrapping, and I can't see many benefits we get from that

Reply:
i can see only one:

having an easy way to validate private keys from elm. that's something we don't do today, we only check if the length for the word list is correct. but there are actually reserved words and forgotten characters

in any way my suggestion was just to keep this opaque types pattern we've been using

Reply:
I went to search for something, and found [this type](https://github.com/cambiatus/frontend/blob/f5c52eb84b6dd32cbc09ee7efa012a7199e79645/src/elm/Eos/Account.elm#L173-L174) that already exists. Should we use that?

Just to be sure, by private key you mean the [hashed analogue of the passphrase](https://github.com/cambiatus/frontend/blob/f5c52eb84b6dd32cbc09ee7efa012a7199e79645/src/elm/Auth.elm#L19-L21)?

Does it make sense to validate either the passphrase (12 words) (other than the word count) or the private key? They're both generated automatically, not from the user :thinking:

Reply:
> I went to search for something, and found this type that already exists. Should we use that?
I think so yeah, in any case we can have a different function that only takes model, this way you don't need to wrap, unwrap every time!

> Just to be sure, by private key you mean the hashed analogue of the passphrase?

> Does it make sense to validate either the passphrase (12 words) (other than the word count) or the private key? They're both generated automatically, not from the user 🤔

Actually the 12 words is a mnemonic that serves as base (with other information to generate the private key)

Reply:
> Actually the 12 words is a mnemonic that serves as base (with other information to generate the private key)

Ok, but does it make sense to validate any of these? We're dealing with

- 12 words (generated for the user, doesn't really make sense to validate)
- Private key (generated for the user, doesn't really make sense to validate)
- PIN (just a sequence of digits, not much to validate other than length and being only digits)

The only data we get from the user is from registration (name, username, email, etc), but I haven't gone through that in this PR

Reply:
> 12 words (generated for the user, doesn't really make sense to validate)
although its generated for they user have to type it on the text area, or paste it to login.

> Private key (generated for the user, doesn't really make sense to validate)
+1

> PIN (just a sequence of digits, not much to validate other than length and being only digits)
+1 Pin validation that we already have is enough